sbin/send_email.py

```diff
@@ -132,14 +133,12 @@ def construct_mail_client():
 
 
 def get_email_queue(db):
-    """Get an active queue of emails waiting to be sent."""
-    result = db.execute(
-        "SELECT emails.id, emails.user_id, users.user_email, emails.subject," +
-        " emails.body FROM emails INNER JOIN users ON" +
-        " emails.user_id = users.user_id WHERE" +
-        " emails.sent is NULL AND emails.error = ''" +
-        " ORDER BY id LIMIT 100;")
-
+    """Get an active queue of internal emails waiting to be sent."""
+    query = """SELECT id, user_id, email_address, subject, body FROM emails
+    WHERE email_address SIMILAR TO :format AND sent is NULL AND
+    error = '' ORDER BY id LIMIT 100;"""
```

Comment:
Have you done any performance measuring on this query to see what the effects of adding `SIMILAR TO` would do to performance?

Reply:
I just created a script to run the old one vs the new one 100 times and average the results. It seems this new script with the SIMILAR TO takes twice as long to run the query. The alternative would be to create another column will a boolean if this is an internal or external address and check the string in PHP. What do you think I should do here? Or do you have a different idea?

Reply:
When doing and discussing stress testing, it's important to provide concrete numbers in all this. How big was the table that you used when running the queries? Have you tried at different sizes so like 1000 rows, 10000 rows, 10,000,000 rows, etc., so that we measure both sizes that are within the ballpark of what Submitty might be facing, but also significantly more so we understand how this might scale.

When you say that the average time doubled, what does that mean? Did it double from taking 10ms to 20ms or 30s to 60s or 1.5min to 3min, etc.

Reply:
Alright I went into more depth with my testing. Here are the results:

1,000 emails: 0.00035 normal and 0.00070 advanced
2,000 emails: 0.00070 normal and 0.0030 advanced
10,000 emails: 0.00065 normal and 0.00120 advanced
100,000 emails: 0.00070 normal and 0.0013 advanced

These results are in seconds. It seems the time stays about the same the more entries I put in. There is of course a limit of 100 so that probably is doing it. It seems that the advanced query takes about double the time but they are both a really small amount of time.

Reply:
Alright, that's probably fine for performance, can re-address if we notice issues in production.